uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Service/CiviHRStatsGatherer.php

```diff
@@ -0,0 +1,172 @@
+  private function setBaseUrl(CiviHRStatistics $stats) {
+    $config = & CRM_Core_Config::singleton();
+    $stats->setSiteUrl($config->userFrameworkBaseURL);
+  }
+
+  /**
+   * Fetches counts and sets them for all required entities
+   *
+   * @param CiviHRStatistics $stats
+   * @throws CiviCRM_API3_Exception
+   */
+  private function setEntityCounts(CiviHRStatistics $stats) {
+    $taskAssignmentsKey = 'uk.co.compucorp.civicrm.tasksassignments';
+    $leaveAndAbsenceKey = 'uk.co.compucorp.civicrm.hrleaveandabsences';
+    $recruitmentKey = 'org.civicrm.hrrecruitment';
+
+    $stats->setEntityCount('contact', $this->getEntityCount('Contact'));
+
+    if (ExtensionHelper::isExtensionEnabled($taskAssignmentsKey)) {
+      $stats->setEntityCount('assignment', $this->getEntityCount('Assignment'));
+      $stats->setEntityCount('task', $this->getEntityCount('Task'));
+      $stats->setEntityCount('document', $this->getEntityCount('Document'));
+    }
+
+    if (ExtensionHelper::isExtensionEnabled($leaveAndAbsenceKey)) {
+      // leave requests in last 100 days
+      $format = 'Y-m-d H:i:s';
+      $oneHundredDaysAgo = (new \DateTime('today - 100 days'))->format($format);
+      $params = ['from_date' => ['>=' => $oneHundredDaysAgo]];
+      $last100DaysCount = (int) civicrm_api3('LeaveRequest', 'getcount', $params);
+      $stats->setEntityCount('leaveRequestInLast100Days', $last100DaysCount);
+
+      // total leave requests
+      $leaveRequestCount = $this->getEntityCount('LeaveRequest');
+      $stats->setEntityCount('leaveRequest', $leaveRequestCount);
+    }
+
+    if (ExtensionHelper::isExtensionEnabled($recruitmentKey)) {
+      $stats->setEntityCount('vacancy', $this->getEntityCount('HRVacancy'));
+    }
+
+    $stats->setEntityCount('drupalUser', $this->getEntityCount('UFMatch'));
+  }
+
+  /**
+   * Gets the number of a certain entity in the system.
+   *
+   * @param string $entity
+   *
+   * @return int
+   */
+  private function getEntityCount($entity) {
+    return (int) civicrm_api3($entity, 'getcount');
+  }
+
+  /**
+   * Fetches contact subtypes
+   *
+   * @param CiviHRStatistics $stats
+   * @throws CiviCRM_API3_Exception
+   */
+  private function setContactSubtypes(CiviHRStatistics $stats) {
+    $params = ['parent_id' => ['IS NULL' => 1]];
+    $contactTypes = civicrm_api3('ContactType', 'get', $params)['values'];
+    foreach ($contactTypes as $contactType) {
+      $name = $contactType['name'];
+      $count = civicrm_api3('Contact', 'getcount', ['contact_type' => $name]);
+      $stats->setContactSubtypeCount($name, (int) $count);
+    }
+  }
+
+  /**
+   * Fetches report configurations
+   *
+   * @param CiviHRStatistics $stats
+   */
+  private function setReportConfigurations(CiviHRStatistics $stats) {
+    // Reports are only available in Drupal
+    if (!$this->cmsVariableService instanceof DrupalVariableService) {
+      return;
+    }
+
+    $query = db_select('reports_configuration', 'rc')->fields('rc');
```

Comment:
I know, I started looking at how this could be extracted, maybe with some sort of `CMSDatabase` class that runs queries or fetches rows, but it was going to get complicated and since `report_configurations` is only available in Drupal I decided to keep it simple here and just do a check if it is Drupal before this code.